utils/src/tagged_signing.rs

```diff
@@ -122,16 +129,123 @@ pub trait TaggedKeyPair: Sized {
     where
         TSignable: TaggedSignable,
     {
-        let data = self.sign(&message.message_data());
+        let signature = self.sign(&message.message_data());
 
-        TaggedSignature::from_bytes(data)
+        TaggedSignature::from_bytes(signature)
     }
 }
 
 pub trait TaggedPublicKey {
     fn verify(&self, msg: &[u8], sig: &[u8]) -> bool;
 }
 
+#[derive(Clone)]
+pub struct TaggedSigned<TSignable, TScheme>
+where
+    TSignable: TaggedSignable,
+    TScheme: TaggedKeyPair,
+{
```

Comment:
`Deserialize`, `Serialize` could probably be derived we add a `Tag<TSignable>` field that always serializes to a `u8`, would probably be easier and more future-proof.

Reply:
I added a `u8` field initialized with `TSignable::TAG` in the `new` function. Let me know if you were thinking on a better solution